### ISSUE TITLE
Check if test in(linked_edges) can be replaced with test in(training_set) everywhere

### ISSUE THREAD

**Issue by rfhh** [ielhelw/MCMC_FOR_AMMSB#9]
Save training set alongside held-out set, test set, and aux stuff.
Check whether the application still needs the full graph alongside the training set -- hopefully not.

**Comment by rfhh** [comment]
_Almost_ everywhere where in(linked_edges) occurs, it is accompanied by tests for held_out_set and test_set membership, either directly, or by previous sampling like the minibatch: the minibatch edges are not in held_out_set or test_set.
There is only one site which (in WenZhe's code) directly checks for membership of linked_edges: in update_phi(_nodes). Ismail, did you say this is a bug in the machine learning algorithm?

**Comment by ielhelw** [comment]
The neighbors of every node in the mini-batch are sampled in a way that they are never members of held-out or test set (See MCMCSamplerStochastic::sample_neighbor_nodes()). Therefore, the check in update_phi() needs to check for a neighbors membership in linked_edges only.
